### PR TITLE
TST: Move tests/scripts to scripts/tests

### PR DESCRIPTION
**Pull request by gfyoung**
Moves the `tests/scripts` directory to `scripts/tests`, as these tests don't really belong in the top-level `pandas` directory (they test the `scripts` directory, not `pandas`).

### CODE REVIEW

File: scripts/tests/test_validate_docstrings.py [hunk 1]
```diff
@@ -1,8 +1,10 @@
-import os
-import sys
-
-import numpy as np
 import pytest
+import string
+import random
```
Comment (datapythonista) [inline scripts/tests/test_validate_docstrings.py]: standard library imports should go before third-party module imports

https://www.python.org/dev/peps/pep-0008/#imports

File: scripts/tests/test_validate_docstrings.py [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
+import numpy as np
+
+validate_docstrings = pytest.importorskip("validate_docstrings")
```
Comment (datapythonista) [inline scripts/tests/test_validate_docstrings.py]: I guess I'm missing something obvious, but why is this needed? can't we simply `import validate_docstrings`?

File: scripts/tests/test_validate_docstrings.py [hunk 3]
```diff
@@ -60,9 +62,9 @@ def random_letters(self):
         letters : str
             String of random letters.
         """
-        length = random.randint(1, 10)  # noqa: F821
-        letters = ''.join(random.choice(string.ascii_lowercase)  # noqa: F821
-                          for i in range(length))
+        length = random.randint(1, 10)
+        letters = "".join(random.choice(string.ascii_lowercase)
+                          for _ in range(length))
```
Comment (datapythonista) [inline scripts/tests/test_validate_docstrings.py]: I know this comes from the original code, but we don't need a comprehension here, we can simply do `''.join(random.sample(string.ascii_lowercase, length))`

**Review by datapythonista — APPROVED** [review]
Good changes, I think they make things simpler and better organized. Thanks!

File: scripts/tests/test_validate_docstrings.py (hunk 2 above)
Comment (gfyoung) [reply scripts/tests/test_validate_docstrings.py]: That's vestigial paranoia from when we skipped tests if we couldn't see `validate_docstrings`.  I agree that we can just import this without the protection.

File: .travis.yml [hunk 4]
```diff
@@ -57,6 +57,9 @@ matrix:
     - dist: trusty
       env:
         - JOB="3.6, coverage" ENV_FILE="ci/travis-36.yaml" TEST_ARGS="--skip-slow --skip-network" PANDAS_TESTING_MODE="deprecate" COVERAGE=true DOCTEST=true
+    - dist: trusty
```
Comment (jreback) [inline .travis.yml]: i don't think we need another build

File: scripts/tests/conftest.py [hunk 5]
```diff
@@ -0,0 +1,3 @@
+def pytest_addoption(parser):
```
Comment (jreback) [inline scripts/tests/conftest.py]: what is this for?

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by WillAyd** [comment]
Think we also need to update the Makefile to ensure this still gets linted (?)

File: .travis.yml (hunk 4 above)
Comment (jreback) [reply .travis.yml]: you can just put on an existing build

**Review by jreback — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 4 above)
Comment (gfyoung) [reply .travis.yml]: That's fair, and I agree we can do that.  I don't have access to my computer ATM (which is why I haven't addressed the comments).

File: scripts/tests/conftest.py (hunk 5 above)
Comment (gfyoung) [reply scripts/tests/conftest.py]: As it says in the code: it's needed to placate `setup.cfg`.

File: scripts/tests/test_validate_docstrings.py (hunk 1 above)
Comment (gfyoung) [reply scripts/tests/test_validate_docstrings.py]: Indeed, they should!  Done.

File: scripts/tests/test_validate_docstrings.py (hunk 3 above)
Comment (gfyoung) [reply scripts/tests/test_validate_docstrings.py]: Fair enough.  Done.

**Comment by gfyoung** [comment]
> Think we also need to update the Makefile to ensure this still gets linted (?)

We don't use the `Makefile` when doing the linting on Travis.  That being said, I don't see why we don't add it as another command.

**Comment by gfyoung** [comment]
@jreback @datapythonista @WillAyd : I've addressed all of your comments, and everything is still green!  PTAL.

**Review by jreback — APPROVED** [review]

**Comment by gfyoung** [comment]
@jreback : Is this good to merged since you approved it?

**Comment by jreback** [comment]
yep
